httpclient5/src/main/java/org/apache/hc/client5/http/SystemDefaultDnsResolver.java

```diff
@@ -40,7 +40,14 @@ public class SystemDefaultDnsResolver implements DnsResolver {
 
     @Override
     public InetAddress[] resolve(final String host) throws UnknownHostException {
-        return InetAddress.getAllByName(host);
+        try {
```

Comment:
@arturobernalg This approach has a substantial drawback: DNS lookup failures completely unrelated to LLv6 will not generate two DNS queries. For some applications it may be not a big deal but for some it may. What is the problem with always calling `#stripsIPv6ZoneId` prior to calling `InetAddress#getAllByName`?

Reply:
The reason why we don't want to call stripsIPv6ZoneId all the time is that it involves additional processing and may not be necessary for hostnames that are not LLv6. This additional processing can cause a slight delay in the overall execution time of the resolve method.
In addition, if we always call stripsIPv6ZoneId before calling InetAddress.getAllByName, it means that we will be making two DNS queries for every hostname, even those that are not LLv6. This can result in unnecessary network traffic and may cause performance issues in certain environments.
Therefore, the approach of only calling stripsIPv6ZoneId when a UnknownHostException is caught and the hostname is an LLv6 address is a more targeted and efficient solution.

Reply:
@arturobernalg Repeating the same DNS query with the same input makes no sense and a lot of folks will be upset due to unnecessary execution costs. Please make sure the second query gets executed only if anything got stripped and the input is not the same

Reply:
@ok2c That's a valid point.
Changed.
TY

Reply:
@arturobernalg I suggest `#stripsIPv6ZoneId` be inlined as it is not being used anywhere else and extra hostname comparison be eliminated.